mozjs/js/rust/src/magicdom/domquad.rs

```diff
@@ -20,6 +25,22 @@ magic_dom! {
     }
 }
 
+impl NumSlots for DOMQuad {
+    const NUM_SLOTS: u32 = <dompoint::DOMPoint as NumSlots>::NUM_SLOTS_PTR * 4;
```

Comment:
OK, that makes sense. We could use better names, though. Even different traits, I think:

```rust
pub trait NumSlots {
    const NUM_SLOTS: u32;
}

pub trait InheritanceSlots {
    const INHERITANCE_SLOTS: u32;
}
```

something like that?